src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobContentId.cs

```diff
@@ -77,6 +77,24 @@ public static unsafe BlobContentId FromHash(byte[] hashCode)
             guidPtr[7] = (byte)((guidPtr[7] & 0x0f) | (4 << 4));
             guidPtr[8] = (byte)((guidPtr[8] & 0x3f) | (2 << 6));
 
+            // the code above assumed the host is little-endian; if this is not the case,
+            // we need to fix up the first three fields (int, short, short)
+            if (!BitConverter.IsLittleEndian)
+            {
+                byte tmp = guidPtr[0];
```

Comment:
Can we use `ReverseEndianness` oor `Read*LittleEndian` to better handle this?

Reply:
It appears during CI this code is also built against older .NET versions that do not have the BinaryPrimitives features.  This has caused failures with earlier versions of the PR, which is why I went back to open-coding everything ...